### PR TITLE
Change ExecutionContext.Clone to public

### PR DESCRIPTION
**Pull request by shargon**
It will be required for https://github.com/neo-project/neo/pull/1629

### CODE REVIEW

**Comment by ProDog** [comment]
Is this ready for testing? @shargon

**Comment by shargon** [comment]
> Is this ready for testing? @shargon

Yes please

**Comment by ProDog** [comment]
Can you give me an example? what it is used for, or how to use it :thinking: @shargon

**Comment by erikzhang** [comment]
I think it is not necessary. See https://github.com/neo-project/neo/pull/1629#issuecomment-625688951.

We need to load a new context for the callbacks.

**Comment by shargon** [comment]
> We need to load a new context for the callbacks.

But then we wont have the shared variables

**Comment by erikzhang** [comment]
Callback methods should be stateless.

**Comment by ProDog** [comment]
I have tested it whit https://github.com/neo-project/neo-devpack-dotnet/pull/262.
- The callback function runs normally without arguments.
- An error occurred while compiling the contract, when the callback function has arguments:

image

In addition, an exception is thrown when deploy contract if `CreateFuncPointer()` is public:

image

**Comment by shargon** [comment]
> Callback methods should be stateless.

But how can we prevent the use of static variables?

**Comment by shargon** [comment]
@ProDog could you create this test in dev-pack repository?

**Comment by ProDog** [comment]
> @ProDog could you create this test in dev-pack repository?

OK

File: src/neo-vm/ExecutionEngine.cs [hunk 1]
```diff
@@ -1303,11 +1303,17 @@ private bool HandleException()
...
         }
 
-        public ExecutionContext LoadClonedContext(int initialPosition)
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public ExecutionContext LoadClonedContext(int initialPosition, int rvcount = 0)
+        {
+            return LoadClonedContext(CurrentContext, initialPosition, rvcount);
+        }
+
+        public ExecutionContext LoadClonedContext(ExecutionContext context, int initialPosition, int rvcount = 0)
```
Comment (erikzhang) [inline src/neo-vm/ExecutionEngine.cs:1312]: Can we remove this and use `LoadContext()`?

Comment (shargon) [reply src/neo-vm/ExecutionEngine.cs:1312]: Yes, If we change `Clone` to public

**Pull request merged**